mlrun/feature_store/retrieval/job.py

```diff
@@ -45,25 +46,50 @@ def run_merge_job(
     name = f"{name}_merger"
     run_config = run_config or RunConfig()
     kind = run_config.kind or ("spark" if engine == "spark" else "job")
+    run_config.kind = kind
+    default_code = _default_merger_handler.replace("{{{engine}}}", merger.__name__)
     if not run_config.function:
         function_ref = vector.spec.function.copy()
         if function_ref.is_empty():
             function_ref = FunctionReference(name=name, kind=kind)
         if not function_ref.url:
-            function_ref.code = _default_merger_handler.replace(
-                "{{{engine}}}", merger.__name__
-            )
+            function_ref.code = default_code
         run_config.function = function_ref
 
     function = run_config.to_function(kind, merger.get_default_image(kind))
+
+    function.with_code(body=default_code)
```

Comment:
Since you're always modifying the function code, I see no point in using the user-provided `handler` later on (in `function.run()`). As discussed, I'm really ok with always using our built-in handler since this is meant to run a very specific code, but let's document it in the docstring for `get_offline_features` and note that if you provide a `RunConfig`, it will be used for configuring runtime resources, locality and type but the `handler` will be overridden.